superset/db_engine_specs.py

```diff
@@ -235,14 +235,15 @@ def where_latest_partition(
 
     @classmethod
     def select_star(cls, my_db, table_name, schema=None, limit=100,
-                    show_cols=False, indent=True, latest_partition=True):
+                    show_cols=False, indent=True, latest_partition=True,
+                    cols=None):
         fields = '*'
-        cols = []
-        if show_cols or latest_partition:
+        cols = cols or []
+        if (show_cols or latest_partition) and not cols:
             cols = my_db.get_table(table_name, schema=schema).columns
 
         if show_cols:
-            fields = [my_db.get_quoter()(c.name) for c in cols]
+            fields = [my_db.get_quoter()(c.get('name')) for c in cols]
```

Comment:
Why did this line change?

Reply:
Good catch, turns out I had inconsistent ways to get the column information here.